lib/show_n_tell_web/talks_resolver.ex

```diff
@@ -0,0 +1,26 @@
+defmodule ShowNTellWeb.TalksResolver do
+  @moduledoc """
+  This module is used from Absinthen GraphQL schema
+  to resolve DB queries and mutations.
+  """
+
+  alias ShowNTell.Repo
+  alias ShowNTell.Talk
+
+  def talk(_root, args, _info) do
+    {:ok, Repo.get(Talk, args[:id])}
+  end
+
+  def talks(_root, _args, _info) do
+    talks = Talk.list_talks()
+
+    {:ok, talks}
+  end
+
+  def create_talk(_root, args, _info) do
+    case Talk.create_talk(args) do
+      {:ok, talk} -> {:ok, talk}
+      _error -> {:error, "could not create talk"}
```

Comment:
Is it valid for a resolver to return a hash? We could do something like this:

```elixir
case Talk.create_talk(args) do
  {:ok, talk} -> {:ok, talk}
  {:error, changeset} -> {:error, ShowNTell.ChangesetView.render("error.json", %{changeset: changeset})}
end
```

`ShowNTell.ChangesetView.render("error.json", %{changeset: changeset})` would return something like:

```
%{errors: %{date: ["can't be blank"], description: ["can't be blank"],
    estimated_duration: ["can't be blank"], title: ["can't be blank"]}}
```

But it would require us to create the ChangesetView module:

```elixir
defmodule ShowNTell.ChangesetView do
  use ShowNTellWeb, :view

  def translate_errors(changeset) do
    Ecto.Changeset.traverse_errors(changeset, &translate_error/1)
  end

  def render("error.json", %{changeset: changeset}) do
    %{errors: translate_errors(changeset)}
  end
end
```

What do you think? I can create a separate PR if you want.

Reply:
Great, I'll create a separate PR for this. :)